Update of the Server Development doc about how 'User Storage Providers' must adhere to Quarkus. (#20512)
keycloak/keycloak: docs/documentation/server_development/topics/user-storage/javaee.adoc

=== Leveraging Jakarta EE

> Since its version 20, Keycloak relies only on Quarkus. And unlike WildFly, Quarkus is not an Application Server. Read more about this at https://www.keycloak.org/migration/migrating-to-quarkus#_quarkus_is_not_an_application_server .

Reviewer: ```suggestion
Since version 20, Keycloak relies only on Quarkus. Unlike WildFly, Quarkus is not an Application Server. For more detail, see https://www.keycloak.org/migration/migrating-to-quarkus#_quarkus_is_not_an_application_server .
```

Author: Suggestion accepted.

Thanks, @andymunro !

> Therefore, the User Storage Providers cannot be packaged within any Jakarta EE component or make it an EJB, as it was possible when Keycloak ran over WildFly in the previous versions.

Reviewer: ```suggestion
Therefore, the User Storage Providers cannot be packaged within any Jakarta EE component or make it an EJB as was the case when Keycloak ran over WildFly in previous versions.
```

Author: Suggestion accepted.

Thanks, @andymunro !

Providers implementations are required to be plain java objects which implement the suitable User Storage SPI interfaces, as was explained in the previous sections. And they must be packaged and deployed as stated in this Migration Guide:

  - https://www.keycloak.org/migration/migrating-to-quarkus#_migrating_custom_providers

> But it is still feasible to implement your custom `UserStorageProvider` class which is able to integrate to an external database via JPA Entity Manager, as shown in this example:

Reviewer: ```suggestion
You can still implement your custom `UserStorageProvider` class, which is able to integrate an external database by JPA Entity Manager, as shown in this example:
```
